### DISCUSSION TITLE
First class support for custom breakpoint in styled component blocks

### DISCUSSION THREAD

**Discussion by LinnJS** [ben-rogerson/twin.macro#439] [Q&A]
For me this is not the end of the world, but it would be really nice if I could use custom breakpoints form the `tailwind.config.js` in a styled component block without having to import them.

This is what I was doing anyways before tailwind but since I have gone down the rabbit hole... the more I learn the more I want. Maybe in a couple months sticking classes right on the HTML will rub off I just don't like to clutter my HTML.

With that being said I would also like for my CSS to get more terse. I have looked around and I see some pretty cool ways to add these props to the theme prop in styled components so I would not have to import it.

But I wanted to open this thread just to ask if having the short cuts directly in the code blocks for Styled Components is a realistic ask at all. I saw this discussion opened up a while back [here](https://github.com/ben-rogerson/twin.macro/issues/6) just wanted to see what was possible.

Thanks in advance for making this package and maintaining it for the community.

**Comment by ben-rogerson** [comment]
Hey there, would the upcoming [screen import](https://github.com/ben-rogerson/twin.macro/pull/408) work for you?
